collect_app/src/main/java/org/odk/collect/android/storage/StorageInitializer.java

```diff
@@ -0,0 +1,41 @@
+package org.odk.collect.android.storage;
+
+import android.os.Environment;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.application.Collect;
+
+import java.io.File;
+
+import timber.log.Timber;
+
+public class StorageInitializer {
+    /**
+     * Creates required directories on the SDCard (or other external storage)
+     *
+     * @throws RuntimeException if there is no SDCard or the directory exists as a non directory
+     */
+    public void createODKDirs() throws RuntimeException {
+        if (!new StorageStateProvider().isStorageMounted()) {
```

Comment:
I think it'd be nice to pull this out as a constructor argument. For me that helps to signal that this is a dependency for the object and I think we'll want to move to having these objects constructed in Dagger to allow us to mock/fake them in tests. For the moment you could also just have a no arg constructor as well like you have in `StoragePathProvider`.

Reply:
Done.